contribution

BilalAlam173 opened the pull request.

BilalAlam173 committed:
login+signup pages

mashhoodr commented:
@afirdousi available for your review. This just has static login and signup pages we will add Firebase and link them up in the next step. Since these pages are isolated, they wont effect rest of the website at the moment.

mashhoodr commented:
@BilalAlam173 why did you remove the fsevents from package.json?

BilalAlam173 commented:
@mashhoodr  fsevent was automaticaly removed in package-lock.json.I will add it again and create another pull request.

mashhoodr commented:
oh its package-lock - @afirdousi was this something you used specifically on your end only? Since its not in package.json

BilalAlam173 committed:
login and signup connected with database

BilalAlam173 commented:
@mashhoodr I have connected both signup and login pages with the database, both are working, I have not done any work on the callback of login as per your instruction, successful login only logs the user detail in the console.

mashhoodr commented:
@afirdousi awaiting review.

mohuk commented on server.js:
```diff
@@ -8,19 +8,32 @@ const db = require('./server/db/db');
 
+//to allow Cross origin requests
+app.use(function(req, res, next) {
+  res.setHeader("Access-Control-Allow-Origin", "*");
```
`"Access-Control-Allow-Origin", "*"` don't allow this. This header shouldn't be there in the first place, use some kind of reverse proxy to mitigate `CORS`

BilalAlam173 replied:
https://www.npmjs.com/package/cors-anywhere can I use this as an alternate?

mohuk replied:
You might not need anything such as this package

mohuk replied:
`CORS` is not required. For development environment use proxying present in `angular-cli`. We can discuss the deployment with @afirdousi

BilalAlam173 replied:
I understand now. Thanks

afirdousi replied:
I agree to @mohuk. I don't think we would need CORS.

mohuk commented on src/app/login/login.component.html:
```diff
@@ -0,0 +1,33 @@
+            <div class="col-md-4">
+            </div>
+            <div class="col-md-4">
+                <form role="form" action="javascript:;" method="post" id="login-form" autocomplete="off">
```
`action="javascript:;" method="post"`

I believe we need to convert this into an `ngForm`

BilalAlam173 replied:
sure i will, in my next commit.

mohuk commented on src/app/services/user.service.ts:
```diff
@@ -0,0 +1,55 @@
+  }
+  
+
+  private handleError (error: Response | any) {
```
Error handling can be improved. `handleError` method will need to be copied where ever required. We can move this into central error handlers or interceptors.

mohuk commented on src/app/signup/signup.component.html:
```diff
@@ -0,0 +1,96 @@
+        <div class="buffer">
+        </div>
+        <div class="row">
+            <form role="form" action="javascript:;" method="post" id="login-form" autocomplete="off">
```
same form issues :(

mohuk commented on src/app/signup/signup.component.ts:
```diff
@@ -0,0 +1,46 @@
+      console.log(data);
+    });
+  }
+
```
:unamused: whitespace

mohuk commented:
@mashhoodr If we are thinking to move to Firebase we might not need all this. We can simply use available authentication options in Firebase

afirdousi commented:
Thanks a bunch for the amazing effort @BilalAlam173. I think this is a good first step moving forward.

My comment ( @mashhoodr @mohuk ) :

- I think we can start with this and later move to `firebase`? I understand firebase is super easy to configure but I am sure **more** developers have used simple `mongoose` with `mongo` on node than firebase (at the moment - maybe I am wrong about devs there ). And if my assumption is correct, we can start with this so more folks can contribute. We will sure have a req open for next quarter to move stuff to firebase. I know it will be redundant effort but may be more learning for new folks who are looking to contribute? 

Sounds good? If @mohuk & @mashhoodr agree, @BilalAlam173 we will need to add a simple profile page and redirect the user to it after both successful sign-in and sign up. I can provide you a simple sketch for a super simple profile page. It will have basic user info + a list of meetups user has signed up for. This meetup portion can be added later after we work on meetup resources section.

mashhoodr commented:
@afirdousi yes, even though this way we have a greater overhead on deployment, db migrations and management - this should do it for now. 

@afirdousi profile page will be a new issue. You can create and add details for that.

The core thing remaining in this will be setting the session on the client side once this is done we can merge this and close the ticket.

afirdousi commented:
@BilalAlam173 can you work on user session in coordination with @mashhoodr and we will be good to go. I will merge the PR.

afirdousi commented:
This PR addresses [Issue#7](https://github.com/angular-pakistan/ngpakistan.com/issues/7)

BilalAlam173 commented:
@afirdousi thanks, yes I will start working on user session once I get those issue fixed pointed out by @mohuk in my next commit. @mashhoodr  can give me the instructions on user sessions, I will start working on it next.

BilalAlam173 committed:
CORS fix

CORS removal and addition of proxy
to use proxy run command : ng serve --proxy-config proxy.config.json

BilalAlam173 committed:
regular forms converted to ngForm

BilalAlam173 committed:
error handling improved

BilalAlam173 committed:
whitespaces removed

afirdousi commented on src/app/services/error.service.ts:
```diff
@@ -0,0 +1,22 @@
+  constructor() { }
+
+   handleError (error: Response | any) {
+    // In a real world app, we might use a remote logging infrastructure
```
Well, its almost a real-world app isn't it 😏 ? jk ... we should definitely log errors remotely in the future.

afirdousi commented:
Since we do not have auto-deployment / CI setup, ill setup it manually soon.
